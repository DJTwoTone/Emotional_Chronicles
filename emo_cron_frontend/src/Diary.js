import React, { useState } from 'react';
import { useHistory } from 'react-router-dom';

import FeelingsCloud from './FeelingsCloud';

import Container from 'react-bootstrap/Container';
import Form from 'react-bootstrap/Form';
import Button from 'react-bootstrap/Button';

function Diary () {

    const history = useHistory()
    const [entry, setEntry] = useState("");

    const [feelings, setFeelings] = useState([]);
    // const [prompt, setPrompt] = useState('');
    // const [inspiration, setInspiration] = ('');


    //word cloud
    //set feelings in feelings
    //prompt changing
    //inspitation change
    //change prompt
    //handle change
    //handle submit

    function handleChange(evt) {
        setEntry(evt.target.value);
    }

    async function handleSubmit(evt) {
        evt.preventDefault();
        console.log('entry:', entry, 'feelings:', feelings)
    }


    return (
        <Container className='justify-content-center m-5'>
            <FeelingsCloud className='m-5' feelings={feelings} setFeelings={setFeelings} />
            <Form className='m-5'>
                <Form.Group controlId='diaryEntry'>
                    <Form.Control 
                          as='textarea'
                          type='text'
                          name='diaryEntry'
                          value={entry}
                          onChange={handleChange}
                          rows={20}
                    />

                </Form.Group>
            </Form>
        </Container>

    )




}


export default Diary;