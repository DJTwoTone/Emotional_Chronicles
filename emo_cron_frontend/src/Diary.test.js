import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Diary from './Diary';

it('should render', function() {
    render(
        <MemoryRouter>
            <Diary />
        </MemoryRouter>
    );
})

it('should match the snapshot', function() {
    const { asFragment } = render(
        <MemoryRouter>
            <Diary />
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
})