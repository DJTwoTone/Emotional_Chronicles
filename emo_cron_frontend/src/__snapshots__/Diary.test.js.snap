// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="justify-content-center my-5 container-fluid"
  >
    <div
      class="shadow m-4 p-4 bg-dark"
    >
      <p
        class="text-light m-3"
      >
        click any of these emoptions you are feeling today:
      </p>
      <div
        class="tag-cloud"
      />
    </div>
    <div
      class="m-4 shadow card"
    >
      <div
        class="card-body"
      >
        <p
          class="display-5 lead card-text"
        >
          Something to write about...
        </p>
        <p
          class="display-4 card-text"
        />
        <button
          class="btn btn-dark"
          type="button"
        >
          GET A NEW PROMPT
        </button>
      </div>
    </div>
    <form
      class="m-4 shadow"
    >
      <div
        class="form-group"
      >
        <div
          id="pattern"
        >
          <textarea
            class="form-control"
            id="diaryEntry"
            name="diaryEntry"
            rows="20"
            type="text"
          />
        </div>
      </div>
      <button
        class="mb-4 btn btn-dark"
        type="submit"
      >
        RECORD
      </button>
    </form>
    <div
      class="m-4 shadow card"
    >
      <div
        class="card-body"
      >
        <div
          class="card-header"
        >
          Something to inspire you...
        </div>
        <p
          class="display-4 card-text"
        />
        <button
          class="mb-4 btn btn-dark"
          type="button"
        >
          GET MORE INSPIRATION
        </button>
        <div
          class="card-footer"
        >
          <p
            class="card-text"
          >
            Share some inspiration...
          </p>
          <form
            class="form-inline"
          >
            <input
              class="col-10 form-control"
              name="inspText"
              type="text"
              value=""
            />
            <button
              class="ml-2 btn btn-dark"
              type="submit"
            >
              SHARE
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
